File: WEB2/mongoDb.js
const MongoClient = require('mongodb').MongoClient;

// Conexión a la base de datos
const uri = "mongodb://localhost:27017/";
const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });

// Nombre de la base de datos y colección
const dbName = "sample_airbnb";
const collectionName = "trips";

// Obtener la colección
async function allRegistros(client) {
    try {
        // Conectar a la base de datos
        await client.connect();

        // Obtener la colección "listingsAndReviews"
        const db = client.db(dbName);
        const collection = db.collection(collectionName);

        // Contar los registros que tengan 2, 3, 4 o 5 camas
        const count = await collection.countDocuments({
            $or: [{ beds: 2 }, { beds: 3 }, { beds: 4 }, { beds: 5 }],
        });

        // Imprimir el resultado
        console.log("Resultado:", count);
    } catch (error) {
        console.error(error);
    } finally {
        // Cerrar la conexión a la base de datos
        await client.close();
    }
}

// Llamar a la función
allRegistros(client);

File: WEB2/mongoDb.py
import pymongo

# Conexión a la base de datos
client = pymongo.MongoClient("mongodb://localhost:27017/")

# Nombre de la base de datos y colección
dbname = "sample_airbnb"
collection_name = "trips"

# Obtener la colección
collection = client.db(dbname).collection(collection_name)

# Función para obtener todos los registros
async def allRegistros(client):
    # Obtener la colección "listingsAndReviews"
    db = client.db("sample_airbnb").collection("listingsAndReviews")

    # Contar los registros que tengan 2, 3, 4 o 5 camas
    result = await db.find({"$or": [{"beds": 2}, {"beds": 3}, {"beds": 4}, {"beds": 5}]}).count()

    # Imprimir el resultado
    print("Resultado", result)

# Llamar a la función
allRegistros(client)
